rinex/src/context/mod.rs

```diff
@@ -24,561 +22,268 @@ use rinex_qc_traits::HtmlReport;
+    RinexMergeError(#[from] RinexMergeError),
     #[error("failed to extend sp3 context")]
-    SP3MergeError(#[from] sp3::MergeError),
+    SP3MergeError(#[from] SP3MergeError),
 }
 
-#[derive(Default, Debug, Clone)]
-pub struct ProvidedData<T> {
-    /// Source paths
-    pub paths: Vec<PathBuf>,
-    /// Data
-    pub data: T,
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Hash)]
+pub enum ProductType {
+    /// GNSS carrier signal observation in the form
+    /// of Observation RINEX data.
+    Observation,
+    /// Meteo sensors data wrapped as Meteo RINEX files.
+    MeteoObservation,
+    /// Broadcast Navigation message as contained in
+    /// Navigation RINEX files.
+    BroadcastNavigation,
+    /// High precision clock data wrapped in Clock RINEX files.
+    HighPrecisionOrbit,
+    /// High precision orbital attitudes wrapped in Clock RINEX files.
+    HighPrecisionClock,
+    /// Antenna calibration information wrapped in ANTEX special RINEX files.
+    Antex,
+    /// Precise Ionosphere state wrapped in IONEX special RINEX files.
+    Ionex,
 }
 
-impl<T> ProvidedData<T> {
-    /// Returns reference to Inner Data
-    pub fn data(&self) -> &T {
-        &self.data
-    }
-    /// Returns mutable reference to Inner Data
-    pub fn data_mut(&mut self) -> &mut T {
-        &mut self.data
-    }
-    /// Returns list of files that created this context
-    pub fn paths(&self) -> &[PathBuf] {
-        &self.paths
+impl From<RinexType> for ProductType {
+    fn from(rt: RinexType) -> Self {
+        match rt {
+            RinexType::ObservationData => Self::Observation,
+            RinexType::NavigationData => Self::BroadcastNavigation,
+            RinexType::MeteoData => Self::MeteoObservation,
+            RinexType::ClockData => Self::HighPrecisionClock,
+            RinexType::IonosphereMaps => Self::Ionex,
+            RinexType::AntennaData => Self::Antex,
+        }
     }
 }
 
-/// RnxContext is a structure dedicated to RINEX post processing workflows,
-/// like precise timing, positioning or atmosphere analysis.
-#[derive(Default, Debug, Clone)]
-pub struct RnxContext {
-    /// Optional OBS RINEX: provides GNSS signals, allows precise positioning.
-    pub obs: Option<ProvidedData<Rinex>>,
-    /// Optional NAV RINEX: allows precise positioning.
-    pub nav: Option<ProvidedData<Rinex>>,
-    /// Optional CLK RINEX
-    pub clk: Option<ProvidedData<Rinex>>,
-    /// Optional ATX RINEX
-    pub atx: Option<ProvidedData<Rinex>>,
-    /// Optional SP3 Orbits. Allows precise positioning.
-    pub sp3: Option<ProvidedData<SP3>>,
-    /// Optional Meteo file. Provides detailed modeling and meteo surveying.
-    pub meteo: Option<ProvidedData<Rinex>>,
-    /// Optional IONEX for accurate Ionosphere compensation.
-    pub ionex: Option<ProvidedData<Rinex>>,
+#[derive(Clone)]
+enum BlobData<'a> {
+    /// RINEX content
+    Rinex(&'a Rinex),
+    /// SP3 content
+    Sp3(&'a SP3),
 }
```

Comment:
@lnicola 

This is the new structure trying to remove the need to Clone internal structures